File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cpp
#include <unordered_set>
#include <vector>
class Solution {
public:
    ListNode* modifiedList(std::vector<int>& nums, ListNode* head) {
        std::unordered_set<int> nums_set(nums.begin(), nums.end());
        ListNode dummy(-1);
        dummy.next = head;
        ListNode* prev = &dummy;
        ListNode* current = head;
        while (current) {
            if (nums_set.find(current->val) != nums_set.end()) {
                prev->next = current->next;
            } else {
                prev = current;
            }
            current = current->next;
        }
        return dummy.next;
    }
};

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def modifiedList(self, nums, head):
        """
        :type nums: List[int]
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        temp=head
        prev=None
        numset=set(nums)
        while temp:
            if head.val in numset:
                head=head.next
            else:
                temp=None
        temp=head.next
        prev=head
        while temp:
            if(temp.val in numset):
                prev.next=temp.next
                
                if(temp.next is None):
                    break
            else:
                prev=temp
            temp=temp.next
        return head

            
        